src/cpp/core/include/core/Random.hpp

```diff
@@ -16,33 +16,32 @@
 #ifndef CORE_RANDOM_HPP
 #define CORE_RANDOM_HPP
 
+#ifndef _WIN32
+# include <unistd.h>
+#else
+# include <process.h>
+#endif
+
 #include <limits>
 #include <ctime>
 
 #include <boost/random.hpp>
 
+#include <core/system/System.hpp>
+
 namespace rstudio {
 namespace core {
 namespace random {
 
 template <typename T>
-T uniformRandomInteger()
+T uniformRandomInteger(
+      T minValue = std::numeric_limits<T>::min(),
+      T maxValue = std::numeric_limits<T>::max())
 {
-  // setup generator and distribution
-  typedef boost::mt19937 GeneratorType;
-  typedef boost::uniform_int<T> DistributionType;
-  GeneratorType generator(std::time(nullptr));
-  DistributionType distribution(std::numeric_limits<T>::min(),
-                                std::numeric_limits<T>::max());
-
-  // create variate generator
-  boost::variate_generator<GeneratorType, DistributionType> vg(generator,
-                                                               distribution);
-
-  // return random number
-  return vg();
+   boost::random::mt19937 gen(getpid() + std::time(NULL));
```

Comment:
We include `::getpid()` here just to provide a little bit of extra certainty that the starting seed should be different. This should be sufficient for desktop, at least -- and shouldn't hurt for server configurations (although this API doesn't appear to be used outside of desktop)

Reply:
I recognize this particular use case doesn't have to be "secure" in the sense that port is not guessable, we are just trying to avoid collissions. But since this is in our core utils, how about if we use: std::chrono::high_resolution_clock::now().time_since_epoch().count() (which is "clock ticks") in place of std::time(NULL), that I think is in seconds.  I don't see any other uses of this util that require "secure random" - if so, we'd have to do a lot more work!